packages/stable/src/types.ts

```diff
@@ -1656,6 +1656,124 @@ export interface ProjectUpdate {
   authentication?: InputProjectAuthentication;
 }
 
+/**
+* Change that will be applied to the array.
+*/
+export interface ArrayPatchString {
+    set?: Set<string>;
+    remove?: Set<string>;
+    add?: Set<string>;
+}
+
+export interface OidcConfigurationUpdate {
+    modify?: OidcConfigurationUpdateDTO;
+    set?: OidcConfigurationDTO;
+    setNull?: boolean;
+}
+
+export interface ClaimName {
+    claimName?: string;
+}
+
+export interface ModifyPatchInteger {
+    set?: number;
+    setNull?: boolean;
+}
+
+export interface OidcConfigurationUpdateDTO {
+    jwksUrl?: SetField<string>;
+    tokenUrl?: SinglePatchString;
+    issuer?: SetField<string>;
+    audience?: SetField<string>;
+    skewMs?: ModifyPatchInteger;
+    accessClaims?: ModifyPatchClaimNames;
+    scopeClaims?: ModifyPatchClaimNames;
+    logClaims?: ModifyPatchClaimNames;
+}
+
+export interface ModifyPatchClaimNames {
+    set?: Array<ClaimName>;
+    add?: Array<ClaimName>;
+    remove?: Array<ClaimName>;
+}
+
+export interface ProjectUpdateDTO {
+    update: ProjectUpdateObjectDTO;
+}
+
+/**
+* Contains the instructions on how to update the project. Note: azureADConfiguration, oidcConfiguration and oAuth2Configuration are mutually exclusive
+*/
+export interface ProjectUpdateObjectDTO {
+    name?: SinglePatchRequiredString;
+    defaultGroupId?: SinglePatchLong;
+    validDomains?: ArrayPatchString;
+    applicationDomains?: ArrayPatchString;
+    authenticationProtocol?: SinglePatchRequiredString;
+    azureADConfiguration?: SinglePatchAzureADConfiguration;
+    oAuth2Configuration?: SinglePatchOAuth2Configuration;
+    oidcConfiguration?: OidcConfigurationUpdate;
+}
+
+export interface OidcConfigurationDTO {
+    /**
+    * The URL where the signing keys used to sign tokens from the identity provider are located
+    */
+    jwksUrl: string;
+    /**
+    * The URL of the OAuth 2.0 token endpoint
+    */
+    tokenUrl?: string;
+    /**
+    * The expected issuer value
+    */
+    issuer: string;
+    /**
+    * The expected audience value (for CDF)
+    */
+    audience: string;
+    /**
+    * Permitted clock skew (ms)
+    */
+    skewMs?: number;
+    /**
+    * Which claims to link CDF groups to, in order to grant access
+    */
+    accessClaims: Array;
+    /**
+    * Which claims to use when scoping access granted by access claims
+    */
+    scopeClaims: Array;
+    /**
+    * Which token claims to record in the audit log
+    */
+    logClaims: Array;
+}
+
+export interface SinglePatchAzureADConfiguration {
+    set: AzureADConfiguration;
+    setNull: boolean;
+}
+
+/**
+* Set a new value for the string, or remove the value.
+*/
+export interface SinglePatchLong {
+    set?: number;
+    setNull?: boolean;
+}
+
+export interface SinglePatchOAuth2Configuration {
+    set?: OAuth2Configuration;
+    setNull?: boolean;
+}
```

Comment:
Try to use/reuse generic union types, something like `SinglePatchString` but maybe with a generic type?

Reply:
Using optional parameters in this case is less preferred as `set` and `setNull` exclude each other